appscale/tools/agents/azure_agent.py

```diff
@@ -1193,3 +1214,92 @@ def does_resource_group_exist(self, resource_group_name, resource_client):
     if resource_group_name in resource_group_names:
       return True
     return False
+
+  def create_image(self, instance_id, image_name, parameters, dest_container):
```

Comment:
My basis for having used this structure was the EC2 agent, which has a create_image function that appscale-testing uses.